src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,455 @@
+
+        function createVariableDeclarationOrAssignment(prevArgName: [Identifier, number], rightHandSide: Expression): NodeArray<Statement> {
+            if (prevArgName[1] > 1) {
+                prevArgName[1] -= 1;
+                return createNodeArray([createStatement(createAssignment(getSynthesizedDeepClone(prevArgName[0]), rightHandSide))]);
+            }
+
+            prevArgName[1] -= 1;
+            return createNodeArray([createVariableStatement(/*modifiers*/ undefined,
+                (createVariableDeclarationList([createVariableDeclaration(getSynthesizedDeepClone(prevArgName[0]), /*type*/ undefined, rightHandSide)], NodeFlags.Let)))]);
+        }
+
+        function subtractReferences(pos: number, argName: [Identifier, number]): void {
+            const refNode = argName[0].parent ? argName[0] : argName[0].original;
+            if (hasArgName && refNode) {
+                const numArgUses = FindAllReferences.getReferenceEntriesForNode(pos, refNode, context.program, [context.sourceFile], context.cancellationToken);
+                if (numArgUses) {
+                    argName[1] -= numArgUses.length;
+                }
+            }
+        }
+
+        const hasPrevArgName = prevArgName && prevArgName[0].text.length > 0;
+        const hasArgName = argName && argName[0].text.length > 0;
+        const nextDotThen = lastDotThenMap.get(String(getNodeId(parent)));
+        switch (func.kind) {
+            case SyntaxKind.Identifier:
+                if (!hasArgName) {
+                    break;
+                }
+
+                const synthCall = createCall(getSynthesizedDeepClone(func) as Identifier, /*typeArguments*/ undefined, [argName[0]]);
+                if (!nextDotThen) {
+                    return createNodeArray([createReturn(synthCall)]);
+                }
+
+                if (!hasPrevArgName) {
+                    break;
+                }
+
+                return createVariableDeclarationOrAssignment(prevArgName!, createAwait(synthCall));
+
+            case SyntaxKind.FunctionDeclaration:
+            case SyntaxKind.FunctionExpression:
+            case SyntaxKind.ArrowFunction:
+
+                if (isFunctionLikeDeclaration(func) && func.body && isBlock(func.body) && func.body.statements) {
+                    const innerRetStmts = getReturnStatementsWithPromiseCallbacks(func.body);
+                    const innerCbBody = getInnerCallbackBody(checker, innerRetStmts, synthNamesMap, lastDotThenMap, context, prevArgName);
+
+                    subtractReferences(func.body.statements.pos, argName);
+
+                    if (innerCbBody.length > 0) {
+                        return createNodeArray(innerCbBody);
+                    }
+
+                    return nextDotThen ? removeReturns(func.body.statements, prevArgName![0]) : getSynthesizedDeepClones(func.body.statements);
+                }
+                else if (isArrowFunction(func)) {

```

Comment:
Looks like this should be a separate `case` instead of an `if` ? Or add a comment if this is intentional